### PR TITLE
feat: tile padding prop codemod

### PR DESCRIPTION
**Pull request by sjohnson-sage**
### Proposed behaviour

Add `replace-prop-value` codemod which can replace the value for a prop for any component.

Add `tile-update-padding-prop` codemod which replaces the `padding` prop on `Tile` to be `p` and updates the values as below: 

"XS" -> 1 
"S" -> 2
"M" -> 3
"L" -> 4
"XL" -> 5

### Current behaviour

These codemods do not currently exist.

### Checklist

- [x] Release notes (Conventional Commits) 
- [x] Unit tests
- [ ] Readme updated

### CODE REVIEW

**Review by mkrds — APPROVED** [review]

File: bin/cli.js [hunk 1]
```diff
@@ -147,6 +154,35 @@ Example
       runTransform(target, command, program, { component, prop })
     );
 
+  program
```
Comment (jamime) [inline bin/cli.js:157]: This failed to transform the following code 

```jsx
<Button renderRouterLink={(props) => <Link {...props} />} to="/protected">
      This link is protected
</Button>
```

`npx carbon-codemod replace-prop-value . carbon-react/lib/components/button to "/protected" "/private"`

```
npx carbon-codemod replace-prop-value . carbon-react/lib/components/button to "/protected" "/private"  
jscodeshift --verbose=2 --ignore-pattern=**/node_modules/** --transform /Users/jamie/carbon-codemod/transforms/replace-prop-value/replace-prop-value.js /Users/jamie/app-1 --component=carbon-react/lib/components/button --prop=to --oldValue=/protected --newValue=/private
```

I thought it might be the leading `/` but removing it had no effect.

File: bin/cli.js [hunk 2]
```diff
@@ -147,6 +154,35 @@ Example
...
+    .command(
+      "replace-prop-value <target> <component> <prop> <oldValue> <newValue>"
+    )
+    .description(
+      `
+Codemod used for prop value changing
+Usage
+  npx carbon-codemod replace-prop-value <target> <component> <prop> <oldValue> <newValue>
```
Comment (jamime) [inline bin/cli.js:165]: We should also cater for named imports, e.g. `import {TileFooter} from ...`

**Review by jamime — CHANGES_REQUESTED** [review]

File: bin/cli.js (hunk 2 above)
Comment (sjohnson-sage) [reply bin/cli.js:165]: It already would wouldn't it? I'm not really sure what you mean?

Comment (sjohnson-sage) [reply bin/cli.js:165]: It's just looking for import path, it doesn't matter if it's named or not

**Review by mkrds — APPROVED** [review]

File: bin/cli.js (hunk 1 above)
Comment (jamime) [reply bin/cli.js:157]: This now runs, but the replacement is ends up being wrapped in a `JSXExpressionContainer` instead of being a string.

```diff
-- <Button renderRouterLink={(props) => <Link {...props} />} to="/protected">
++ <Button renderRouterLink={(props) => <Link {...props} />} to={"/private"}>
```

However, there might be some cases where we need the container, e.g. when the user is trying to set the prop to be a number. We potentially have the same problem for the existing value, users might be trying to match only strings, or only numbers.

We'll need a way to determine this, we can either use an optional argument
e.g. 

```
npx carbon-codemod replace-prop-value ... 1 2 --replaceWithContainer
```

or we can use a mix of `parseInt/parseFloat` to work out if the argument is numeric, if it is we will treat it as a number and put it in a container. Users could then escape this by passing escaped quotes. e.g.

```
npx carbon-codemod replace-prop-value ... \"1\" \"2\"
```

We'd have to handle single and double quotes.

Although the second solution sounds messy, I think it's probably the better solution.

Commander has some options which allows us to coerce the value before it's passed to the codemod which I think will make things simpler. In the codemod we could then rely on the `typeof/isNaN`

https://github.com/tj/commander.js/#custom-option-processing

File: bin/cli.js (hunk 2 above)
Comment (jamime) [reply bin/cli.js:165]: Did you create a ticket to do this?

Comment (sjohnson-sage) [reply bin/cli.js:165]: I did indeed, FE-3229

File: bin/cli.js (hunk 1 above)
Comment (sjohnson-sage) [reply bin/cli.js:157]: My idea for this was put everything in a container and it'll always work correctly, and the formatting for strings should be handled by formatters in consuming apps. It won't always be but at least a string inside a container is a valid prop

If an app is running prettier on commit for example, it would just remove the brackets for us here

Comment (sjohnson-sage) [reply bin/cli.js:157]: I don't actually see what you mean about the same problem for the existing value? I'll message you on slack about it

File: transforms/replace-prop-value/README.md [hunk 3]
```diff
@@ -54,6 +54,10 @@ If there is a pattern that you use that is not transformed, please file a featur
 
 `npx carbon-codemod replace-prop-value <target> <component-import-path> <prop> <old-value> <new-value>`
 
+Note: To use a string as `old-value` or `new-value`, format it as `\"string\"` on the command line, i.e.:
```
Comment (jamime) [inline transforms/replace-prop-value/README.md:57]: I think we need to update this, because the example below doesn't need to be escaped. It's only if you want to use a numeric value as a string that you need to quote it.

**Review by jamime — CHANGES_REQUESTED** [review]

File: bin/cli.js [hunk 4]
```diff
@@ -147,6 +154,35 @@ Example
...
+
+  target       Files or directory to transform
+  component    Import path of component which prop should be renamed
+  attribute    Prop name
```
Comment (jamime) [inline bin/cli.js:169]: Why was this renamed from `prop` to `attribute`? L165 still cals it prop.

Comment (sjohnson-sage) [reply bin/cli.js:169]: Because I'd called it `prop` here and `attribute` in `replace-prop-value.js` which meant it wasn't getting passed through initially so I had to change it in one place or the other. In `replace-prop-value.js` it's being passed into the `replaceAttributeValue` function so I decided just to call it attribute

I'll update that on line 165!

**Review by jamime — DISMISSED** [review]

**Review by jamime — APPROVED** [review]

**Review by mkrds — APPROVED** [review]
